### PR TITLE
Started work on course directore screen

### LINKED ISSUE #27: Create CourseDirectory Screen Draft

### LINKED ISSUE #30: Fix scrollbar color
Need to look into how to change the scrollbar color into something different. From experimenting this turned out to be controlled by accentColor, however changing that results in braking the button texts, as those are also controlled by this color.
Ideally branch from branch in  #29, to be able to merge it into this easily.

### PR DESCRIPTION
**Pull request by michalrzak**

### CODE REVIEW

**Comment by michalrzak** [comment]
Please somebody pull this and try if the scrollbar in the course directory shows in an emulator. It is not showing in the web view and I have found some articles online that suggest that scrollbars don't show in the web view. @peter-gy @memoriesadrift

**Comment by michalrzak** [comment]
This pr will be ready to merge once the scrollbar color gets fixed #30

**Comment by michalrzak** [comment]
This branch is now ready to merge into main. As it covers a lot of changes I would like for both of you to take a look at it. (at least briefly)

File: lib/model/modelTranslate.dart
Comment (peter-gy) [inline lib/model/modelTranslate.dart]: I would prefer to handle the internationalization of the app using a consumer-provider pattern. I have already created an issue for this. See #33.

Comment (michalrzak) [reply lib/model/modelTranslate.dart]: For now I can change the Translate classes to return I18NString instead of string and define an instance variable until this feature is implemented. Once it is there I can change it to use the provider.
Would this be sufficient?

Comment (peter-gy) [reply lib/model/modelTranslate.dart]: I think let's just keep it like this for now, we can make the necessary changes as soon as the providers are set up.

File: lib/screens/CourseDirectoryScreen.dart
Comment (peter-gy) [inline lib/screens/CourseDirectoryScreen.dart:8]: The course directory is basically the root of a module tree, therefore the ideal solution would be to create a single widget that represents a module and handle the recursive rendering of its submodules. The base case of the recursion would be when a module has no further modules. In this case, the courses/exams of that module could be rendered.

The code that you wrote gives a great foundation to this! I think it would be a viable solution to refactor `StudyModuleScreen` to use a single widget, `StudyModuleView` and reuse it in `StudyModuleScreen` and `CourseDirectoryScreen` as well.

Comment (michalrzak) [reply lib/screens/CourseDirectoryScreen.dart:8]: I will look into this before merging and try to push a solution ASAP.

File: lib/screens/StudyModuleScreen.dart
Comment (memoriesadrift) [inline lib/screens/StudyModuleScreen.dart]: Is there any reason we don't use context.of here to directly get the parent width?

File: lib/model/modelTranslate.dart
Comment (memoriesadrift) [reply lib/model/modelTranslate.dart]: I can handle internationalisation in a separate issue once this and other UI/UX issues are closed and the templates are created.

File: lib/screens/CourseDirectoryScreen.dart
Comment (memoriesadrift) [inline lib/screens/CourseDirectoryScreen.dart]: nitpick but let's rename `StudyModuleTranslate` to something else. It evokes almost a method name for me, let's go with `StudyModuleTranslated` or `StudyModuleInternationalised` or something similar.

**Review by memoriesadrift — APPROVED** [review]

File: lib/screens/StudyModuleScreen.dart
Comment (michalrzak) [reply lib/screens/StudyModuleScreen.dart]: Honestly just because I don't know how that works haha. Could you give me a code snippet so I can change it?

File: lib/screens/CourseDirectoryScreen.dart
Comment (michalrzak) [reply lib/screens/CourseDirectoryScreen.dart]: Maybe the choice of name really is poor as it is not supposed to handle translating. It's more of an abstraction layer between the actual `StudyModule`, which contains optionals and the actual app, to make accessing the widgets a bit easier.

Maybe `StudyModuleWrap` or `StudyModuleAbstract`?

File: lib/screens/CourseDirectoryScreen.dart
Comment (michalrzak) [reply lib/screens/CourseDirectoryScreen.dart:8]: Done.

File: lib/screens/CourseDirectoryScreen.dart
Comment (memoriesadrift) [reply lib/screens/CourseDirectoryScreen.dart]: Maybe `StudyModuleDisplay` if it handles displayable data?

File: lib/screens/StudyModuleScreen.dart
Comment (memoriesadrift) [reply lib/screens/StudyModuleScreen.dart]: Keeping it your way seems very valid too! Check this thread for alternatives
https://stackoverflow.com/questions/41558368/how-can-i-layout-widgets-based-on-the-size-of-the-parent

Comment (michalrzak) [reply lib/screens/StudyModuleScreen.dart]: In that case I think I will keep it like this. I think with the comment it is pretty self explanatory what is happening there and to me it seems a bit clearer then fiddling with some sort of widget wrap.

File: lib/screens/CourseDirectoryScreen.dart
Comment (michalrzak) [reply lib/screens/CourseDirectoryScreen.dart]: Hmmm I don't think I like that too much. It does not handle displaying the data rather just preparing it for displaying.

`StudyModuleParsed` maybe?

Comment (michalrzak) [reply lib/screens/CourseDirectoryScreen.dart]: `StudyModuleWrapped` <- final name

**Pull request merged**
